### PR TITLE
Wp 468

### PR DESCRIPTION
**Pull request by paveldhq**

### CODE REVIEW

File: inc/Smartling/Submissions/SubmissionEntity.php [hunk 1]
```diff
@@ -75,6 +80,7 @@ class SubmissionEntity extends SmartlingEntityAbstract
         self::SUBMISSION_STATUS_IN_PROGRESS,
         self::SUBMISSION_STATUS_COMPLETED,
         self::SUBMISSION_STATUS_FAILED,
+        self::SUBMISSION_STATUS_CANCELLED,
```
Comment (dimitrystd) [inline inc/Smartling/Submissions/SubmissionEntity.php]: I remember i asked to call it Canceled but just checked [submissions service](https://wiki.smartling.net/display/DEV/Submission+service+architecture#Submissionservicearchitecture-Translationsubmissionsstates) and it uses Deleted. On the other hand, [connectors commons](https://github.com/Smartling/connector-commons/blob/8308354895b735ad3ba07a89fc3a87ddca25bc0c/commons-core/src/main/java/com/smartling/connectors/hosted/core/submission/SubmissionStatus.java#L9) use Cancelled. Even not sure what is better.

File: inc/Smartling/Helpers/SubmissionCleanupHelper.php [hunk 2]
```diff
@@ -249,18 +249,20 @@ private function processDeletion(array $searchParams)
...
             $this->getLogger()
                 ->debug(vsprintf('No submissions found for search params: %s', [var_export($searchParams, true)]));
         }
     }
 
-    private function deleteSubmission(SubmissionEntity $submission)
+    private function cancelSubmission(SubmissionEntity $submission)
     {
         $this->unlinkContent($submission);
-        $this->getSubmissionManager()->delete($submission);
```
Comment (dimitrystd) [inline inc/Smartling/Helpers/SubmissionCleanupHelper.php]: It seems we don't use `SubmissionManager.delete()` anymore and can remove this method.

File: inc/Smartling/Submissions/SubmissionEntity.php [hunk 3]
```diff
@@ -67,6 +67,11 @@ class SubmissionEntity extends SmartlingEntityAbstract
      */
     const SUBMISSION_STATUS_FAILED = 'Failed';
 
+    /**
+     * Submission Status 'Cancelled'
+     */
+    const SUBMISSION_STATUS_CANCELLED = 'Cancelled';
```
Comment (dimitrystd) [inline inc/Smartling/Submissions/SubmissionEntity.php]: Can you verify "check translation progress" task that we don't try to get status for canceled submissions?
Also i expected to see changes in download task and probably on UI (don't show Download button and link for canceled submission)

File: inc/Smartling/Helpers/SubmissionCleanupHelper.php [hunk 4]
```diff
@@ -249,18 +249,20 @@ private function processDeletion(array $searchParams)
...
+        $submission->setStatus(SubmissionEntity::SUBMISSION_STATUS_CANCELLED);
+        $submission->setTargetId(0);
+        $submission = $this->getSubmissionManager()->storeEntity($submission);
         $this->deleteFile($submission);
```
Comment (PavelLoparev) [inline inc/Smartling/Helpers/SubmissionCleanupHelper.php]: Do we still need to delete the file?

File: inc/Smartling/Submissions/SubmissionEntity.php (hunk 1 above)
Comment (paveldhq) [reply inc/Smartling/Submissions/SubmissionEntity.php]: So what should we use?

File: inc/Smartling/Helpers/SubmissionCleanupHelper.php (hunk 2 above)
Comment (paveldhq) [reply inc/Smartling/Helpers/SubmissionCleanupHelper.php]: We use it in `Smartling\Services\BlogRemovalHandler->blogRemovalHandler()`

File: inc/Smartling/Helpers/SubmissionCleanupHelper.php (hunk 4 above)
Comment (paveldhq) [reply inc/Smartling/Helpers/SubmissionCleanupHelper.php]: removed

**Pull request merged**